### ISSUE TITLE
Refactor module architecture

### ISSUE THREAD

**Issue by JKrag** [Praqma/Praqmatic-Automated-Changelog#73]
Current:
Decorator has sections for Jira, Fogbugz etc.
Task has sections for Jira, Fogbugz etc. 

Better:
All Jira belongs together
All Fogbugz belongs together

**Comment by buep** [comment]
Agree, and our original intended architecture also says so: https://github.com/Praqma/Praqmatic-Automated-Changelog/blob/master/docs/designview.png

They belong in the task systems implementations.

@MadsNielsen - please estimate such a change, and elaborate any details not obvious regarding implementation.

**Comment by MadsNielsen** [comment]
The good thing is taht the decorator module is totally interchangable, so the only thing that needs to change is the logic being moved to the task module. 

It shouldn't take more than a couple of hours to implement, and we're not adding new functionality, so the existing tests will have to be modified to use the new location of the classes. Since you do Module::Classname. 

I can't see anything not-so-obvious that needst to be accounted for, so a couple of hours is my guess.
